src/WebJobs.Script/Description/BindingMetadata.cs

```diff
@@ -3,14 +3,37 @@
 
 namespace Microsoft.Azure.WebJobs.Script.Description
 {
+    /// <summary>
+    /// Represents the metadata for a function parameter binding.
+    /// </summary>
     public class BindingMetadata
     {
+        /// <summary>
+        /// Gets or sets the name of the binding.
+        /// </summary>
         public string Name { get; set; }
 
+        /// <summary>
+        /// Gets or sets an app setting name for a setting whose value
+        /// contains the connection details that should be used for this binding.
+        /// E.g. for a Queue binding, this might be set to the name of a an app
+        /// setting containing the Azure Storage connection string to use.
+        /// </summary>
+        public string Connection { get; set; }
```

Comment:
I think it's good to have this here. 
It can also drive consistency. Right now, some bindings (like event hub) treat Connection as the literal connection string, and others treat it with 1-level of indirection (ie, an appsetting) to the connection string. 


---
In reply to: [56767402](https://github.com/Azure/azure-webjobs-sdk-script/pull/128#discussion_r56767402) [](ancestors = 56767402)